packages/terra-table/src/SelectableUtils.js

```diff
@@ -16,21 +29,76 @@ const initialSingleSelectRowIndex = (rows) => {
 };
 
 /**
-* Returns if the selected row index is already selected for SingleSelectableRows.
-*/
+ * Returns the indexes of the selected rows for MultiSelectableRows.
+ * To be used in the constructor, to set initial state.
+ */
+const initialMultiSelectRowIndexes = (rows, maxSelectionCount) => {
+  if (!rows || !rows.length) {
+    return [];
+  }
+  // Find the rows which are selected and are selectable
+  const selectedIndexes = [];
+  const validatedMaxSelectionCount = validatedMaxCount(rows, maxSelectionCount);
+  for (let i = 0; i < rows.length; i += 1) {
+    if (selectedIndexes.length >= validatedMaxSelectionCount) {
+      break;
+    }
+    if (rows[i].props.isSelected) {
+      selectedIndexes.push(i);
+    }
+  }
+  return selectedIndexes;
+};
+
+/**
+ * Returns a new array, updated with the newIndex being added or removed from the existing.
+ */
+const updatedMultiSelectedIndexes = (currentIndexes, newIndex) => {
+  let newIndexes = [];
+  if (currentIndexes.length) {
```

Comment:
want to check if currentIndexes is not undefined first

Reply:
Set to if (currentIndexes && currentIndexes.length)